graph/src/components/store.rs

```diff
@@ -683,6 +683,19 @@ pub trait Store: Send + Sync + 'static {
             .map_err(|e| format_err!("Failed to query SubgraphDeployment entities: {}", e))
             .map(|entity_opt| entity_opt.is_some())
     }
+
+    /// Return true if the deployment with the given id is fully synced,
+    /// and return false otherwise. Errors from the store are passed back up
+    fn is_deployment_synced(&self, id: &SubgraphDeploymentId) -> Result<bool, Error> {
+        let filter = EntityFilter::new_equal("id", id.to_string());
+        let entity = self.find_one(SubgraphDeploymentEntity::query().filter(filter))?;
```

Comment:
If we already have the ID it would be cheaper to use store.get instead of find_one.

Reply:
Try `self.get(SubgraphDeploymentEntity::key(id))`

Reply:
Good catch! I ended up having the function take ownership so that callers can decide whether `id` needs to be cloned or not.